### DISCUSSION TITLE
Passing data to loader when calling useLoaderData

### DISCUSSION THREAD

**Discussion by filiplikavcan** [remix-run/remix#4056] [Proposals]
```jsx
export const loader: LoaderFunction = async ({ data }) => {
    return json({
        number: data.number + 1
    })
}

export default function () {
    const data = useLoaderData<{ number: number }>({ number: 1 });

    return <div>{data.number} should be 2.</div>;
}
```

This would allow to do things like `useLoaderData(useOutletContext())` - that is to parametrize loader not only with the request but also with the outlet context.

**Comment by sergiodxa** [comment]
The only way to pass parameters to the loader function is through the URL.

Also useLoaderData doesn't do the fetch itself, the data is fetched outside React and stored in a context, the hook only access that context.

This is the actual code of the hook, as you can see it's just a single line:

https://github.com/remix-run/remix/blob/f4d6ff6aa1ab0ca0af0482d958ad2abc8bb9d0fe/packages/remix-react/components.tsx#L1416-L1418

So it wouldn't be possible to pass parameters to the loader this way.

**Reply by filiplikavcan** [reply]
OK, I understand. Thank you for explanation.

**Reply by ttnbtfy** [reply]
Hi @filiplikavcan how did you do that?
